src/sentry/api/endpoints/project_create_sample_transaction.py

```diff
@@ -0,0 +1,44 @@
+import os
+from datetime import datetime, timedelta
+
+import pytz
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectEventPermission
+from sentry.api.serializers import serialize
+from sentry.constants import DATA_ROOT
+from sentry.utils import json
+from sentry.utils.dates import to_timestamp
+from sentry.utils.samples import create_sample_event_basic
+
+
+class ProjectCreateSampleTransactionEndpoint(ProjectEndpoint):
+    # Members should be able to create sample events.
+    # This is the same scope that allows members to view all issues for a project.
+    permission_classes = (ProjectEventPermission,)
+
+    def post(self, request, project):
+        samples_root = os.path.join(DATA_ROOT, "samples")
+        with open(os.path.join(samples_root, "react-transaction.json")) as fp:
+            data = json.load(fp)
+
+        timestamp = datetime.utcnow() - timedelta(minutes=1)
+        timestamp = timestamp - timedelta(microseconds=timestamp.microsecond % 1000)
+        timestamp = timestamp.replace(tzinfo=pytz.utc)
+        data["timestamp"] = to_timestamp(timestamp)
+
+        start_timestamp = timestamp - timedelta(seconds=3)
+        data["start_timestamp"] = to_timestamp(start_timestamp)
+        for span in data.get("spans", []):
+            # Use data to generate span timestamps consistently and based
+            # on event timestamp
+            duration = span.get("data", {}).get("duration", 10.0)
+            offset = span.get("data", {}).get("offset", 0)
+            span_start = data["start_timestamp"] + offset
+            span["start_timestamp"] = span_start
+            span["timestamp"] = span_start + duration
+
+        event = create_sample_event_basic(data, project.id, raw=True)
+
+        data = serialize(event, request.user)
```

Comment:
@Neo-Zhixing rather than returning the full event, could we just return a dictionary that includes the link to the event?

Reply:
Of course we can, but I'm doing this so that it is consistent with the /create_samples endpoint.

Reply:
@Neo-Zhixing ahhh good point, let's be consistent.